nemoguardrails/library/autoguard/actions.py

```diff
@@ -0,0 +1,496 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from typing import Optional
+
+import aiohttp
+
+from nemoguardrails.actions import action
+
+log = logging.getLogger(__name__)
+
+
+@action(name="call autoguard gender bias api", is_system_action=True)
+async def call_autoguard_gender_bias_api(context: Optional[dict] = None):
+    api_key = os.environ.get("AUTOGUARD_API_KEY")
+
+    if api_key is None:
+        raise ValueError("AUTOGUARD_API_KEY environment variable not set.")
+
+    user_message = context.get("user_message")
+    bot_message = context.get("bot_message")
+    if bot_message:
+        prompt = bot_message
+    else:
+        prompt = user_message
+
+    print(prompt)
+
+    model = context.get("model")
+    url = "http://35.225.99.81:8888/"
```

Comment:
which url is this? @KimiJL this should be specific to an nvidia autoguard service url?

Reply:
as in setting up an independent instance for them? or wait until the SaaS instance is ready? @rahm-decypher

Reply:
resolved in earlier commits